src/app/[locale]/page.tsx

```diff
@@ -27,35 +28,66 @@ export default function Index() {
     lat: 0,
     lng: 0,
   });
+  const [searchCenter, setSetSearchCenter] = useState<{ lat: number; lng: number }>({
```

Comment:
Isn't this [0,0] initialization conflicting with setting the search center coordinates based on the user's IP address? It seems like the search center will always default to the coast of Africa in this case.